### ISSUE TITLE
Semantic markup and containerless vs as-element

### ISSUE THREAD

**Issue by Alexander-Taran** [aurelia-contrib/help-needed#3]
https://discourse.aurelia.io/t/use-of-containerless/817
Is a very nice thread which could answer a lot of questions about semantic markup and styling..
where one has to add bogus rules to make custom element display:block and height:100% just to be able to size it's children..

**Comment by Alexander-Taran** [comment]
@huochunpeng want to try to take a stab at it?
summarize your comments from the thread in some article?

**Comment by huochunpeng** [comment]
Yep. No only for this one. I want to summarize more of my topics from discourse. Bit tight this month, I will be more active after March.

I also have plan to use discourse as a testing ground for my new writings in mind.

Also plan to extract some of my private code to open source plugins.

Also have to plan to redo doc for bcx-aurelia-dnd and bcx-aurelia-reorderable-repeat, plus some small fix. Then I will market it on discourse. Have a look of reorderable-repeat, it is pretty cool.

**Comment by Alexander-Taran** [comment]
@huochunpeng a link attached to the name 'reorderable-repeat' would be so great (-:

**Comment by huochunpeng** [comment]
Was writing on my phone.
https://github.com/buttonwoodcx/bcx-aurelia-reorderable-repeat

**Comment by Alexander-Taran** [comment]
This is cool.. I was needing this one.. had to go with implementing dnd myself.. 
and now i decided to shoot myself in the foot and try to figure out how to reorder items in the grid instead of flexbox (-:

**Comment by huochunpeng** [comment]
You need to go down to bcx-aurelia-dnd or we can create a wrapper lib to reorder grid.
Reorderable-repeat can handle flex layout even with row/column wrapping. Just set the order hint to follow flex direction, read the last feature demonstrated in the doc.

**Comment by Alexander-Taran** [comment]
Yeah I understood that from the docs. Really nice.
